sdk/cosmos/azure-cosmos/test/test_crud_subpartition.py

```diff
@@ -0,0 +1,742 @@
+                                 }
+                                ]
+        created_documents = []
+        for document_definition in document_definitions:
+            created_documents.append(created_collection.create_item(
+                body=document_definition))
+        self.assertEqual(len(created_documents), len(document_definitions))
+
+        # Query all documents should return all items
+        document_list = list(created_collection.query_items(query='Select * from c', enable_cross_partition_query=True))
+        self.assertEqual(len(document_list), len(document_definitions))
+
+        # Query all items with only CA for 1st level. Should return only 8 items instead of 10
+        document_list = list(created_collection.query_items(query='Select * from c', partition_key=['CA']))
+        self.assertEqual(8, len(document_list))
+
+        # Query all items with CA for 1st level and Oxnard for second level. Should only return 3 items
+        document_list = list(created_collection.query_items(query='Select * from c', partition_key=['CA', 'Oxnard']))
+        self.assertEqual(3, len(document_list))
+
+        # Query for specific zipcode using 1st level of partition key value only:
+        document_list = list(created_collection.query_items(query='Select * from c where c.zipcode = "93033"',
+                                                            partition_key=['CA']))
+        self.assertEqual(1, len(document_list))
+
+        # Query Should work with None values:
+        document_list = list(created_collection.query_items(query='Select * from c', partition_key=[None, '93033']))
+        self.assertEqual(0, len(document_list))
+
+        # Query Should Work with non string values
+        document_list = list(created_collection.query_items(query='Select * from c', partition_key=[0xFF, 0xFF]))
+        self.assertEqual(0, len(document_list))
+
+        document_list = list(created_collection.query_items(query='Select * from c', partition_key=[None, None]))
+        self.assertEqual(0, len(document_list))
+
+        document_list = list(created_collection.query_items(query='Select * from c', partition_key=["", ""]))
+        self.assertEqual(0, len(document_list))
+
+        document_list = list(created_collection.query_items(query='Select * from c', partition_key=[""]))
+        self.assertEqual(0, len(document_list))
+
+        # Negative Test, prefix query should not work if no partition is given (empty list is given)
+        try:
+            document_list = list(created_collection.query_items(query='Select * from c', partition_key=[]))
+            self.fail("Test did not fail as expected")
+        except exceptions.CosmosHttpResponseError as error:
+            self.assertEqual(error.status_code, StatusCodes.BAD_REQUEST)
+            self.assertTrue("Cross partition query is required but disabled"
+                            in error.message)
+
+    def test_partition_key_range_overlap(self):
+        Id = 'id'
```

Comment:
@ealsur These tests should also simulate a situation where a EPK range spans multiple physical partitions.